xds/internal/balancer/priority/balancer_priority.go

```diff
@@ -147,83 +145,91 @@ func (pb *priorityBalancer) switchToChild(child *childBalancer, priority int) {
 	// so we don't stop the init timer.
 	// - a high priority is READY, an config update always triggers switch to
 	// it.
-	if pb.childInUse == child.name && child.started {
+	if b.childInUse == child.name && child.started {
 		return
 	}
-	pb.childInUse = child.name
-	pb.priorityInUse = priority
+	b.childInUse = child.name
+	b.priorityInUse = priority
 
 	// Init timer is always for childInUse. Since we are switching to a
 	// different child, we will stop the init timer no matter what. If this
 	// child is not started, we will start the init timer later.
-	if pb.priorityInitTimer != nil {
-		pb.priorityInitTimer.Stop()
-		pb.priorityInitTimer = nil
-	}
+	b.stopPriorityInitTimer()
 
 	if !child.started {
 		child.start()
-		pb.priorityInitTimer = time.AfterFunc(defaultPriorityInitTimeout, func() {
-			pb.mu.Lock()
-			defer pb.mu.Unlock()
-			if pb.childInUse != child.name {
+		// Need this local variable to capture timerW in the AfterFunc closure
+		// to check the stopped boolean.
+		timerW := &timerWrapper{}
+		b.priorityInitTimer = timerW
+		timerW.timer = time.AfterFunc(defaultPriorityInitTimeout, func() {
+			b.mu.Lock()
+			defer b.mu.Unlock()
+			// FIXME: it would be wrong if p is switching from 1 to 0 then to 1.
```

Comment:
Filing a bug is preferable to a comment in the code.

Reply:
This is fixed. Comment deleted.